hedera-node/hedera-app/src/main/java/com/hedera/node/app/tss/TssBaseServiceImpl.java

```diff
@@ -117,7 +129,47 @@ public void bootstrapLedgerId(
     public void setCandidateRoster(@NonNull final Roster roster, @NonNull final HandleContext context) {
         requireNonNull(roster);
         // (TSS-FUTURE) https://github.com/hashgraph/hedera-services/issues/14748
-        tssSubmissions.submitTssMessage(TssMessageTransactionBody.DEFAULT, context);
+
+        // generate TSS messages based on the active roster and the candidate roster
+        final var tssStore = context.storeFactory().writableStore(ReadableTssBaseStore.class);
+        final var maxSharesPerNode =
+                context.configuration().getConfigData(TssConfig.class).maxSharesPerNode();
+        final var sourceRoster =
+                context.storeFactory().readableStore(ReadableRosterStore.class).getActiveRoster();
+        final var activeRosterHash = RosterUtils.hash(sourceRoster).getBytes();
+        final var candidateRosterHash = RosterUtils.hash(roster).getBytes();
+        final var tssPrivateShares =
+                getTssPrivateShares(sourceRoster, maxSharesPerNode, tssStore, candidateRosterHash, context);
+        final var candidateRosterParticipantDirectory = computeTssParticipantDirectory(roster, maxSharesPerNode, (int)
+                context.networkInfo().selfNodeInfo().nodeId());
+
+        int shareIndex = 0;
+        for (final var tssPrivateShare : tssPrivateShares) {
+            final var tssMsg = tssLibrary.generateTssMessage(candidateRosterParticipantDirectory, tssPrivateShare);
+            final var tssMessage = TssMessageTransactionBody.newBuilder()
+                    .sourceRosterHash(activeRosterHash)
+                    .targetRosterHash(candidateRosterHash)
+                    .shareIndex(shareIndex++)
+                    .tssMessage(Bytes.wrap(tssMsg.bytes()))
+                    .build();
+            tssSubmissions.submitTssMessage(tssMessage, context);
+        }
```

Comment:
This is the code I'm implementing in #14748. I'm confused why this is submitting TSS messages. Is there a place in the TSS proposals I can refresh on this?

Reply:
The code you are implementing is setting candidate roster, from the issue and also looked on your branch. That part is correct. 

In addition, the tss messages should be submitted when we set candidate roster. Michael had a fake implementation there and I removed it, since TssMessageHandler is implemented. 

I see the following under `TssCryptographyManager` portion

> If the active roster does not have TSS key material and keyActiveRoster is not true, generate key material for a new ledger id.
Set createNewLedgerId to true
Create a random share for the node.
Send a TssMessageTransaction to the network generated from the random share targeting the candidate roster.

Reply:
Ahh thank you!